src/comp/OtherShareButton.tsx

```diff
@@ -1,25 +1,24 @@
 import React from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { Button, ListItem } from "@material-ui/core";
-import { Icon, IconUrl } from "./shareButtonData/data";
+import { Icon } from "./shareButtonData/data";
 import * as styles from "css/comp/ShareButton.module.css";
 
 type ShareButtonProps = {
-  icons: Array<keyof Icon>;
-  iconsList: Icon;
-  iconsUrl: IconUrl;
+  iconsName: Array<keyof Icon>;
+  icons: Icon;
```

Comment:
```suggestion
  icons: Array<Icon>;
```
わざわざキーの配列を受け取らなくても、
そのまま `Icon` の配列を受け取るだけでいい
`Icon` の名前が必要なら、`Icon` の型定義に `name: string;` を加えた方がいいと思う

Reply:
命名がダメでした、、、 `iconsName` は表示するシェアボタンの種類のリストなので、無いとだめです
`iconsName` -> `shareButtonType` とかでいいですか？

Reply:
いや、表示する種類のリストを受け取るのは冗長だと思います
`Icon` (`ShareSet` みたいな名前の方がいいかもしれない) のリストを受け取るだけにできませんか

Reply:
`ShareButton.tsx` で `map` したら `definition` と `url` だけを渡すことはできると思います

Reply:
そのほうがいいですか？

Reply:
表示に使う情報は、`url` と `definition` とそれ自体の名前だと思うんですけど、
それだけだと足りなくないですか
`Icon` に名前の情報も含まれていればキーの配列を渡さなくて済むと思います

Reply:
理解です

Reply:
修正しました、確認お願いします